src/peft/tuners/vblora/config.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from peft.config import PeftConfig
+from peft.utils import PeftType
+
+
+@dataclass
+class VBLoRAConfig(PeftConfig):
+    """
+    This is the configuration class to store the configuration of a [`VBLoRAConfig`].
+
+    Paper: https://arxiv.org/abs/.
+
+    Args:
+        r (`int`, *optional*, defaults to `4`):
+            VBLoRA parameter dimension ("rank").
+        target_modules (`Union[List[str], str]`):
+            The names of the modules to apply Vera to. Only linear layers are supported.
+        vblora_dropout (`float`):
+            The dropout probability for Vera layers.
+        fan_in_fan_out (`bool`):
+            Set this to True if the layer to replace stores weight like (fan_in, fan_out). For example, gpt-2 uses
+            `Conv1D` which stores weights like (fan_in, fan_out) and hence this should be set to `True`.
+        bias (`str`):
+            Bias type for Vera. Can be 'none', 'all' or 'vera_only'. If 'all' or 'vera_only', the corresponding biases
+            will be updated during training. Be aware that this means that, even when disabling the adapters, the model
+            will not produce the same output as the base model would have without adaptation.
+        modules_to_save (`List[str]`):
+            List of modules apart from Vera layers to be set as trainable and saved in the final checkpoint.
+        init_weights (`bool`):
```

Comment:
In our approach, we shouldn't initialize the additive adaptor to zero because if the vector bank is initialized to zero, both matrices A and B will be zero, resulting in a zero gradient. As a result, we can't align with other methods regarding `init_weights=True`. Do you have any suggestions in this case?

Reply:
I followed your first suggestion in the comment: 
1. I removed `init_weights`.
2. In tests, I set the vector bank to zero to make VB-LoRA as an identity operation. And initialized the vector bank if training is involved in the test.